doc/release-notes-0.4.0.rst

```diff
@@ -0,0 +1,274 @@
+* Updated BSD library:
+
+  * Added API supporting GPS
+  * Fixed various issues (see the :ref:`nrfxlib:bsdlib_changelog`)
+
+* Added sample showing how to use the new GPS API (tested on nRF9160 DK v0.8.5)
+* Added sample showing how to download a new image via HTTP
+* Added multi-image support (see :ref:`ug_pm`)
+
+Release tag
+***********
+
+The release tag for the |NCS| manifest repository (|ncs_repo|) is **v0.4.0**.
+Check the ``west.yml`` file for the corresponding tags in the project repositories.
+
+To use this release, check out the tag in the manifest repository and run ``west update``.
+See :ref:`Getting the nRF Connect SDK code <cloning_the_repositories_win>` for more information.
+
+
+Supported boards
+****************
+
+* PCA10090 (nRF9160 DK) - only for LTE samples
+* PCA10056 (nRF52840 Development Kit)
+* PCA10059 (nRF52840 Dongle)
+* PCA10040 (nRF52 Development Kit)
+* PCA10028 (nRF51 Development Kit)
+* PCA63519 (Smart Remote 3 DK add-on)
+
+
+Required tools
+**************
+
+In addition to the tools mentioned in :ref:`gs_installing`, the following tool versions are required to work with the |NCS|:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Tool
+     - Version
+     - Download link
+   * - SEGGER J-Link
+     - V6.40
+     - `J-Link Software and Documentation Pack`_
+   * - nRF5x Command Line Tools
+     - v9.8.1
+     - `nRF5x Command Line Tools`_
+   * - dtc (Linux only)
+     - v1.4.6 or later
+     - :ref:`gs_installing_tools_linux`
+
+
+As IDE, we recommend to use |SES| (Nordic Edition), version 4.16 or later.
+It is available from the following links:
+
+* `SEGGER Embedded Studio (Nordic Edition) - Windows x86`_
+* `SEGGER Embedded Studio (Nordic Edition) - Windows x64`_
+* `SEGGER Embedded Studio (Nordic Edition) - Mac OS x64`_
+* `SEGGER Embedded Studio (Nordic Edition) - Linux x86`_
+* `SEGGER Embedded Studio (Nordic Edition) - Linux x64`_
+
+
+Change log
+**********
+
+The following sections provide detailed lists of changes by component.
+
+nRF9160
+=======
+
+* Added the following samples:
+
+  * :ref:`http_application_update_sample` - demonstrates a basic FOTA (firmware over-the-air) update
+  * :ref:`mqtt_simple_sample` - connects to an MQTT broker and sends and receives data
+  * :ref:`nrf_coap_client_sample` - receives data from a public CoAP server
+  * Simple GPS sample - gets the current position and logs it on UART
+
+* Renamed the :ref:`secure_partition_manager` sample from Secure Boot to Secure Partition Manager
+* Various fixes and updates to the :ref:`nrfxlib:bsdlib` (in nrfxlib):
+
+  * Updated the library to version 0.3.0 (see the :ref:`nrfxlib:bsdlib_changelog` for details)
+  * Added support for GNSS supporting GPS as a socket
+  * Implemented :cpp:func:`bsd_os_timedwait`, allowing a proper poll operation and blocking sockets
+  * Minor fixes to the nRF91 sockets SHIM layer
```

Comment:
Do we have docs for the nRF91 sockets SHIM layer? (One of the subpages [here](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/bsdlib/doc/api.html)?)

Reply:
No, not really, it's an integration layer between `bsdlib` and Zephyr, namely [this](https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/lib/bsdlib/nrf91_sockets.c) file. A very brief documentation of the API it's implementing is present in Zephyr: https://docs.zephyrproject.org/latest/reference/networking/net_offload.html#socket-offloading

Reply:
OK, wording updated and clarified where the file is located.